.github/workflows/clear_cache.yaml

```diff
@@ -0,0 +1,37 @@
+# This job cleans up the unnecessary caches created on PRs.
+# https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#force-deleting-cache-entries
+name: cleanup caches by a branch
+on:
+  # Only runs on the PRs are closed, regardless of it is merged or not.
+  # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-pull_request-workflow-when-a-pull-request-merges
+  pull_request:
+    types:
+      - closed
+
+jobs:
```

Comment:
this is exactly the same 🍝 https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#force-deleting-cache-entries